workflow/scripts/rNA_flowcells.Rmd

```diff
@@ -387,21 +430,73 @@ cgroups <- as.factor(multiQC$flowcell_lanes)
 cpalette <- brewer.pal(nlevels(cgroups), "Paired")
 
-p <- plot_ly(shared_metadata,
-  x = ~PC1_tin, y = ~PC2_tin, z = ~PC3_tin, color = cgroups, colors = cpalette, hoverinfo = "text", marker = list(size = 8),
-  text = ~ paste(
-    "</br><b> Sample: ", Sample, "</br> Flowcell Lanes: ", flowcell_lanes, "</br> medTIN: ", median_tin, "</b></br></br> % Aligned: ", percent_aligned,
-    "</br> % Dup: ", percent_duplication, "</br> % Coding: ", pct_coding_bases,
-    "</br> % Intronic: ", pct_intronic_bases, "</br></br> medTIN: ", median_tin, "</br> Sequence Range: ", sequence_length,
-    "</br> GC Content: ", gc_content, "</br> Inner Distance Maxima: ", inner_distance_maxima, "</br> Insert Size: ", median_insert_size
-  )
-) %>%
-  add_markers() %>%
-  layout(scene = list(
-    xaxis = list(title = paste0("PC1 (", pc1, "%)")),
-    yaxis = list(title = paste0("PC2 (", pc2, "%)")),
-    zaxis = list(title = paste0("PC3 (", pc3, "%)"))
-  ))
+if (is.na(pc3)) {
+  p <- plot_ly(shared_metadata,
+    x = ~PC1_tin,
+    y = ~PC2_tin,
+    color = cgroups,
+    colors = cpalette,
+    hoverinfo = "text",
+    marker = list(size = 8),
+    text = ~ paste(
+      "</br><b> Sample: ",
+      Sample, "</br> Flowcell Lanes: ",
+      flowcell_lanes, "</br> medTIN: ",
+      median_tin, "</b></br></br> % Aligned: ",
+      percent_aligned,
+      "</br> % Dup: ",
+      percent_duplication, "</br> % Coding: ",
+      pct_coding_bases,
+      "</br> % Intronic: ",
+      pct_intronic_bases, "</br></br> medTIN: ",
+      median_tin, "</br> Sequence Range: ",
+      sequence_length,
+      "</br> GC Content: ",
+      gc_content, "</br> Inner Distance Maxima: ",
+      inner_distance_maxima, "</br> Insert Size: ",
+      median_insert_size
+    )
+  ) %>%
+    add_markers() %>%
+    layout(scene = list(
+      xaxis = list(title = paste0("PC1 (", pc1, "%)")),
+      yaxis = list(title = paste0("PC2 (", pc2, "%)"))
+    ))
+} else {
+  p <- plot_ly(shared_metadata,
+    x = ~PC1_tin,
+    y = ~PC2_tin,
+    z = ~PC3_tin,
+    color = cgroups,
+    colors = cpalette,
+    hoverinfo = "text",
+    marker = list(size = 8),
+    text = ~ paste(
+      "</br><b> Sample: ",
+      Sample, "</br> Flowcell Lanes: ",
+      flowcell_lanes, "</br> medTIN: ",
+      median_tin, "</b></br></br> % Aligned: ",
+      percent_aligned,
+      "</br> % Dup: ",
+      percent_duplication, "</br> % Coding: ",
+      pct_coding_bases,
+      "</br> % Intronic: ",
+      pct_intronic_bases, "</br></br> medTIN: ",
+      median_tin, "</br> Sequence Range: ",
+      sequence_length,
+      "</br> GC Content: ",
+      gc_content, "</br> Inner Distance Maxima: ",
+      inner_distance_maxima, "</br> Insert Size: ",
+      median_insert_size
+    )
+  ) %>%
+    add_markers() %>%
+    layout(scene = list(
+      xaxis = list(title = paste0("PC1 (", pc1, "%)")),
+      yaxis = list(title = paste0("PC2 (", pc2, "%)")),
+      zaxis = list(title = paste0("PC3 (", pc3, "%)"))
+    ))
+}
```

Comment:
As above, I think we should write another function to reduce code duplication here. Actually, we could use one function for both the expression and transcript PCA plots, with arguments for necessary differences in labels/etc.

Reply:
Agreed. latest commit 76fe20f85c1155f8250f8d837c85668f8cbeea8b handles this